lsp/server/src/server.ts

```diff
@@ -120,47 +229,38 @@ async function getSA4UConfig(): Promise<SA4UConfig> {
 	}
 	try {
 		await promises.writeFile('.sa4u/compile_commands.json', readCompileCommands);
-		return new SA4UConfig('.sa4u/');
+		configJSON['CompilationDir'] = '.sa4u/';
+		return new SA4UConfig(configJSON);
 	} catch (err) {
 		console.warn(`Failed to write compile_commands.json into .sa4u directory: ${err}`);
-		return new SA4UConfig(configJSON['CompilationDir']);
+		return new SA4UConfig(configJSON);
 	}
 }
 
-async function getIgnoredFiles(): Promise<SA4UIgnoredFiles> {
-	let readConfig: string;
-	try {
-		readConfig = (await promises.readFile('.sa4u.json')).toString();
-	} catch (err) {
-		console.warn('unable to read SA4U config: ' + err);
-		return new SA4UIgnoredFiles([]);
+async function updateConfigurations(): Promise<void> {
+	if (!existsSync('.sa4u')){
```

Comment:
Instead of using `existsSync` (which blocks the event loop), we should use [`fs.stat`](https://nodejs.org/api/fs.html#filehandlestatoptions) instead.

Reply:
Done

Reply:
Reverted using stat. The server stopped working when stat failed if the file didn't exist.

Reply:
Used promises.access